src/v/container/interval_set.h

```diff
@@ -51,13 +54,27 @@ class interval_set {
      */
     [[nodiscard]] std::pair<const_iterator, bool> insert(interval interval);
 
+    /**
+     * Exposed to make this type serde-able.
+     */
+    std::pair<const_iterator, bool> emplace(key_type k, mapped_type m) {
+        return set_.emplace(std::move(k), std::move(m));
+    }
+
     /**
      * Find the interval containing \p index.
      *
      * If no such interval exists then end() is returned.
      */
     [[nodiscard]] const_iterator find(T index) const;
 
+    /**
+     * Finds the first interval that starts at or is greater than \p index.
+     *
+     * If no such interval exists, then end() is returned.
+     */
+    [[nodiscard]] const_iterator lower_bound(T index) const;
```

Comment:
I think Noah strove for 100% coverage for this class - we should preserve that with a simple test case.